tools/chapel-py/src/chapel/lsp/__init__.py

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright 2024-2024 Hewlett Packard Enterprise Development LP
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from lsprotocol.types import Position, Range
+
+
+def location_to_range(location) -> Range:
+    """
+    Convert a Chapel location into a lsprotocol.types Range, which is
+    used for e.g. reporting diagnostics.
+    """
+
+    start = location.start()
+    end = location.end()
+    # NOTE: we want the char number to be one extra, otherwise single character identifiers don't work right
+    return Range(
+        start=Position(start[0] - 1, start[1] - 1), end=Position(end[0] - 1, end[1])
```

Comment:
Surprised the autoformatter left this as is -- it line wraps in GitHub preview.

Reply:
Default line length is 88. I just reran the formatter with a length of 80, I will commit that shortly.